File: src-tauri/src/commands/overview.rs
use crate::models::result::CommandResult;
use crate::services::overview_service::{self, OverviewStatusData};

#[tauri::command]
pub async fn get_overview_status() -> CommandResult<OverviewStatusData> {
    match overview_service::get_overview_status().await {
        Ok(data) => CommandResult::ok(data),
        Err(error) => CommandResult::err(error),
    }
}

File: src/main/ipc/overview.ts
import { ipcMain } from "electron";
import { IPC } from "../../shared/ipc-channels.js";
import { ok, err } from "../models/result.js";
import { toAppError } from "./utils.js";
import { getOverviewStatus } from "../services/overview-service.js";

export function registerOverviewHandlers(): void {
  ipcMain.handle(IPC.GET_OVERVIEW_STATUS, async () => {
    try { return ok(await getOverviewStatus()); }
    catch (error) { return err(toAppError(error)); }
  });
}
